### ISSUE TITLE
Reelsteady Joiner Error

### ISSUE THREAD

**Issue by 7subaru4** [rubegartor/ReelSteady-Joiner#64]
Hi guys,
Fresh install on my Win10 x64 machine.
I can't make reelsteady joiner to work. All the files are good as it work on my laptop but not on my desktop one...
Any idea what wrong ?

Here is the error :

```
NonError: { error: 'No stream maps found' }
    at module.exports (I:\Program Files\ReelSteady Joiner\resources\app.asar\node_modules\ensure-error\index.js:14:10)
    at handleError (I:\Program Files\ReelSteady Joiner\resources\app.asar\node_modules\electron-unhandled\index.js:50:10)
    at process.<anonymous> (I:\Program Files\ReelSteady Joiner\resources\app.asar\node_modules\electron-unhandled\index.js:130:4)
    at process.emit (node:events:527:28)
    at emit (node:internal/process/promises:140:20)
    at processPromiseRejections (node:internal/process/promises:269:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:97:32)
```

---

reelsteady-joiner 1.3.1
Electron 20.0.2
win32 10.0.19045
Locale: fr

Thanks for your help

**Comment by rubegartor** [comment]
Do you mean that on your laptop ReelSteady Joiner does not give you any errors and on another computer it does? Anyway, without the files giving you the problem I can't offer you any more help.

The error indicates that the files do not have stream maps available, so they lack the necessary information to carry out the entire process.

**Comment by rubegartor** [comment]
Could you share a screenshot of the execution of this command?

`"I:\Program Files\ReelSteady Joiner\node_modules\ffmpeg-static\ffmpeg.exe" -i <mp4_path>`

 A screenshot of each of the chapters would be helpful, as well as from a computer where it works and others where it doesn't, at the same time I insist again that if it is possible to share the files that cause the problem, it would be quite helpful to debug the problem

**Comment by 7subaru4** [comment]
The execution doesn't work as the path is uncorrect.
I tried with ; 

I:\Program Files\ReelSteady Joiner\**resources\app.asar.unpacked**\node_modules\ffmpeg-static\ffmpeg.exe" -i <mp4_path>

I get a blue cmd page for half a second blank and it disapear, same on both machine.
I can send you the files, but I get the same messages with any files I want to join, so not shure if it will help as its not files related but something wrong with my window.

I installed the last ffmpeg, but didn't help

**Comment by rubegartor** [comment]
Correct, the route that I had indicated is wrong although the one you provided should work, expected output:

![image](https://github.com/rubegartor/ReelSteady-Joiner/assets/29512781/a228dd5a-ab60-4707-a227-b10f75f7535e)

Could it be due to some ffmpeg execution problem on your machine?

Something like that coud help (replace to fit your system):
`ffmpeg -i GX010150.MP4 && pause`

The cmd should freeze after executing the ffmpeg command

**Comment by 7subaru4** [comment]
Thanks for the fast reply.
ffmpeg problem I don't know. I update the system variable (path : and add
ffmpeg/bin in the C root with last updated ffmpeg files) but doesn't make
any progress.
Still the same error message from joiner even with a fresh new install.

I am ulploading the files and wil share it to you once its done (4go)
When trying your command with the second part of the video :

**Comment by rubegartor** [comment]
And if you try: I:\Program Files\ReelSteady Joiner\resources\app.asar.unpacked\node_modules\ffmpeg-static\ffmpeg.exe" -i <mp4_path>? Instead of your own installation of ffmpeg.

Reelsteady Joiner doesn't use your ffpmeg instead it uses its own executable included in ReelSteady Joiner, so maybe is due to some error in ReelSteady Joiner ffmpeg.

I'll take a look at the files you attached anyway.

**Comment by 7subaru4** [comment]
> And if you try: I:\Program Files\ReelSteady Joiner\resources\app.asar.unpacked\node_modules\ffmpeg-static\ffmpeg.exe" -i <mp4_path>? Instead of your own installation of ffmpeg.
> 
> Reelsteady Joiner doesn't use your ffpmeg instead it uses its own executable included in ReelSteady Joiner, so maybe is due to some error in ReelSteady Joiner ffmpeg.
> 
> I'll take a look at the files you attached anyway.

This command line just pop up a blue blank window for half a second and that all and disapear forever. 
It does the exact same on my Laptop where RSGO joiner works well.
I installed RSGO joiner on a usb stick. It work when plugged to my laptops but I still have the error message while running it with my desktop computer..
Any idea ?

**Comment by 7subaru4** [comment]
[2023-09-28 22:36:30.022] [info]  OS: win32 - 10.0.19045
[2023-09-28 22:36:30.025] [info]  version: 1.3.1
[2023-09-28 22:36:30.025] [info]  isDev?: false
[2023-09-28 22:36:30.025] [info]  config: {
  "savePath": "I:\\test joiner",
  "exportOption": 0,
  "concurrentProjects": 5
}
[2023-09-28 22:36:30.026] [info]  exePath: C:\Program Files\ReelSteady Joiner
[2023-09-28 22:36:30.026] [info]  macExePath: C:\Program Files\ReelSteady Joiner
[2023-09-28 22:36:30.027] [info]  ffmpegPath: C:\Program Files\ReelSteady Joiner\resources\app.asar.unpacked\node_modules\ffmpeg-static\ffmpeg.exe
[2023-09-28 22:36:30.027] [info]  gyroProcessPath: C:\Program Files\ReelSteady Joiner\app\bin\win\udtacopy.exe
[2023-09-28 22:36:30.035] [info]  exiftool: C:\Program Files\ReelSteady Joiner\app\bin\win\exiftool.exe
[2023-09-28 22:36:30.036] [info]  exiftool config: C:\Program Files\ReelSteady Joiner\app\bin\exiftool.config
[2023-09-28 22:36:30.037] [debug] concat.txt content:
file 'I:\D R O N E S\F P V\A stab JOINER\GX013114.MP4'
file 'I:\D R O N E S\F P V\A stab JOINER\GX023114.MP4'

[2023-09-28 22:36:30.037] [info]  Project: {
  "_id": "7e66c15c-6f71-4deb-a2a3-0fc5e149acd4",
  "_type": "mp4",
  "_projectPath": "I:\\test joiner\\GX013114",
  "_dirPath": "I:\\D R O N E S\\F P V\\A stab JOINER",
  "_files": [
    "GX013114.MP4",
    "GX023114.MP4"
  ],
  "_log": {
    "functions": {},
    "hooks": [],
    "isDev": false,
    "levels": [
      "error",
      "warn",
      "info",
      "verbose",
      "debug",
      "silly"
    ],
    "logId": "7e66c15c-6f71-4deb-a2a3-0fc5e149acd4",
    "variables": {
      "processType": "browser"
    },
    "transports": {}
  },
  "_name": "GX013114",
  "_duration": 324,
  "_modifiedDate": "2023:05:27 22:59:53-07:00",
  "_failed": false,
  "_available": true,
  "_completed": false
}
[2023-09-28 22:36:30.038] [info]  outputName: GX013114_joined.mp4
[2023-09-28 22:36:30.038] [info]  projectPath: I:\test joiner\GX013114
[2023-09-28 22:36:30.038] [info]  concatFilePath: C:\Users\Alexandre W10\AppData\Local\Temp\concat_7e66c15c-6f71-4deb-a2a3-0fc5e149acd4.txt
[2023-09-28 22:36:30.039] [info]  outputFilePath: I:\test joiner\GX013114\GX013114_joined.mp4
[2023-09-28 22:36:30.039] [info]  File sizes:
[2023-09-28 22:36:30.039] [info]  I:\D R O N E S\F P V\A stab JOINER\GX013114.MP4 size: 4010931078B
[2023-09-28 22:36:30.040] [info]  I:\D R O N E S\F P V\A stab JOINER\GX023114.MP4 size: 58567537B

[2023-09-28 22:36:30.103] [debug] Extracted maps: 


**That what I get from the report when it failed to Join.**

**Comment by rubegartor** [comment]
I have generated a debug version that stores some more information in the logs to be able to continue investigating the problem since I have tried to replicate your system and installation but I have not been able to replicate the problem.

If it's not a problem, install the next version and try again, if it still gives you problems, share the log with me again and I will review it for errors.

Debug version: https://we.tl/t-ciAE73QAdO

**Comment by 7subaru4** [comment]
Heya,
Thanks for your help but unfortunately no improvement.
here is the log : 
[2023-09-30 20:41:42.058] [info]  OS: win32 - 10.0.19045
[2023-09-30 20:41:42.060] [info]  version: 1.3.2-debug
[2023-09-30 20:41:42.060] [info]  isDev?: false
[2023-09-30 20:41:42.061] [info]  config: {
  "savePath": "I:\\test joiner",
  "exportOption": 0,
  "concurrentProjects": 5,
  "fileModifyDates": 1,
  "preservePCMAudio": 0
}
[2023-09-30 20:41:42.061] [info]  exePath: C:\Program Files\ReelSteady Joiner
[2023-09-30 20:41:42.061] [info]  macExePath: C:\Program Files\ReelSteady Joiner
[2023-09-30 20:41:42.062] [info]  ffmpegPath: C:\Program Files\ReelSteady Joiner\resources\app.asar.unpacked\node_modules\ffmpeg-static\ffmpeg.exe
[2023-09-30 20:41:42.062] [info]  gyroProcessPath: C:\Program Files\ReelSteady Joiner\app\bin\win\udtacopy.exe
[2023-09-30 20:41:42.070] [info]  exiftool: C:\Program Files\ReelSteady Joiner\app\bin\win\exiftool.exe
[2023-09-30 20:41:42.071] [info]  exiftool config: C:\Program Files\ReelSteady Joiner\app\bin\exiftool.config
[2023-09-30 20:41:42.072] [debug] concat.txt content:
file 'I:\D R O N E S\F P V\Tré la tete JOINER\GX013096.MP4'
file 'I:\D R O N E S\F P V\Tré la tete JOINER\GX023096.MP4'

[2023-09-30 20:41:42.072] [info]  Project: {
  "_id": "34548952-a009-4228-9e14-10d43e5711bf",
  "_type": "mp4",
  "_projectPath": "I:\\test joiner\\GX013096",
  "_dirPath": "I:\\D R O N E S\\F P V\\Tré la tete JOINER",
  "_files": [
    "GX013096.MP4",
    "GX023096.MP4"
  ],
  "_log": {
    "functions": {},
    "hooks": [],
    "isDev": false,
    "levels": [
      "error",
      "warn",
      "info",
      "verbose",
      "debug",
      "silly"
    ],
    "logId": "34548952-a009-4228-9e14-10d43e5711bf",
    "variables": {
      "processType": "browser"
    },
    "transports": {}
  },
  "_name": "GX013096",
  "_outputName": "GX013096_joined.mp4",
  "_duration": 381,
  "_modifiedDate": "2023:04:18 18:28:32-07:00",
  "_failed": false,
  "_available": true,
  "_completed": false
}
[2023-09-30 20:41:42.073] [info]  outputName: GX013096_joined.mp4
[2023-09-30 20:41:42.073] [info]  projectPath: I:\test joiner\GX013096
[2023-09-30 20:41:42.073] [info]  concatFilePath: C:\Users\Alexandre W10\AppData\Local\Temp\concat_34548952-a009-4228-9e14-10d43e5711bf.txt
[2023-09-30 20:41:42.073] [info]  outputFilePath: I:\test joiner\GX013096\GX013096_joined.mp4
[2023-09-30 20:41:42.074] [info]  File sizes:
[2023-09-30 20:41:42.074] [info]  I:\D R O N E S\F P V\Tré la tete JOINER\GX013096.MP4 size: 4010121233B
[2023-09-30 20:41:42.074] [info]  I:\D R O N E S\F P V\Tré la tete JOINER\GX023096.MP4 size: 771231814B

[2023-09-30 20:41:42.106] [debug] Stream-maps:
[2023-09-30 20:41:42.107] [debug] error: Error: Command failed: "C:\Program Files\ReelSteady Joiner\resources\app.asar.unpacked\node_modules\ffmpeg-static\ffmpeg.exe" -f concat -safe 0 -i C:\Users\Alexandre W10\AppData\Local\Temp\concat_34548952-a009-4228-9e14-10d43e5711bf.txt -y
ffmpeg version 6.0-essentials_build-www.gyan.dev Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 12.2.0 (Rev10, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-mediafoundation --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libvpl --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
C:\Users\Alexandre: No such file or directory

[2023-09-30 20:41:42.107] [debug] stdout: 
[2023-09-30 20:41:42.107] [debug] stdout: ffmpeg version 6.0-essentials_build-www.gyan.dev Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 12.2.0 (Rev10, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-mediafoundation --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libvpl --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
C:\Users\Alexandre: No such file or directory

[2023-09-30 20:41:42.108] [debug] Extracted maps:

**Comment by rubegartor** [comment]
Great, now with these new logs I have managed to locate the error, I will fix it for the next version. Thanks for helping!

**Comment by 7subaru4** [comment]
> Great, now with these new logs I have managed to locate the error, I will fix it for the next version. Thanks for helping!

Great, thanks for your help.
Any idea when it will be released ? possibility to send me an beta updated version with the issue fixed ?

**Comment by rubegartor** [comment]
Probably over the next week, I'm doing the last tests to verify that everything is working correctly.

**Comment by rubegartor** [comment]
Fixed in version 1.3.2 now available
